packages/scaffold-config/src/frameworks.ts

```diff
@@ -267,17 +266,53 @@ export const WIZARD_FRAMEWORKS = [
     category: 'library',
     name: 'Svelte.js',
     detectors: [dependencies.WIZARD_DEPENDENCY_SVELTE],
-    supportedBundlers: [dependencies.WIZARD_DEPENDENCY_WEBPACK, dependencies.WIZARD_DEPENDENCY_VITE],
-    dependencies: (bundler: WizardBundler['type'], projectPath: string): Promise<DependencyToInstall[]> => {
-      return Promise.all([
-        getBundlerDependency(bundler, projectPath),
-        isDependencyInstalled(dependencies.WIZARD_DEPENDENCY_SVELTE, projectPath),
-      ])
+    supportedBundlers: ['webpack', 'vite'],
+    dependencies: (bundler: WizardBundler['type']): Cypress.CypressComponentDependency[] => {
+      return [
+        getBundler(bundler),
+        dependencies.WIZARD_DEPENDENCY_SVELTE,
+      ]
     },
     codeGenFramework: 'svelte',
     glob: '*.svelte',
     mountModule: mountModule('cypress/svelte'),
     supportStatus: 'alpha',
     componentIndexHtml: componentIndexHtmlGenerator(),
   },
-] as const
+]
+
+/**
+ * Given a first or third party Component Framework Definition,
+ * resolves into a unified ResolvedComponentFrameworkDefinition.
+ * This way we have a single type used throughout Cypress.
+ */
+export function resolveComponentFrameworkDefinition (definition: Cypress.ComponentFrameworkDefinition | Cypress.ThirdPartyComponentFrameworkDefinition): Cypress.ResolvedComponentFrameworkDefinition {
+  const isThirdPartyDefinition = definition.type.startsWith('cypress-ct-')
+
+  const dependencies: Cypress.ResolvedComponentFrameworkDefinition['dependencies'] = async (bundler, projectPath) => {
+    const declaredDeps = definition.dependencies(bundler)
+
+    // Must add bundler based on launchpad selection if it's a third party definition.
+    if (isThirdPartyDefinition) {
+      declaredDeps.push(getBundler(bundler))
+    }
+
+    return await Promise.all(declaredDeps.map((dep) => isDependencyInstalled(dep, projectPath)))
+  }
+
+  if (isThirdPartyDefinition) {
+    const def = definition as Cypress.ThirdPartyComponentFrameworkDefinition
```

Comment:
Totally up to you, but we could avoid this cast and the extra `def` variable by using a type guard:

```
function isThirdPartyDefinition (definition: Cypress.ComponentFrameworkDefinition | Cypress.ThirdPartyComponentFrameworkDefinition): definition is Cypress.ThirdPartyComponentFrameworkDefinition {
  return definition.type.startsWith('cypress-ct')
}

const isThirdParty = isThirdPartyDefinition(definition)
```

Reply:
Oh nice... let me play with this, but if it's as simple as it looks here, this is better.